includes/admin/load.php

```diff
@@ -262,79 +320,45 @@ function perflab_activate_plugin() {
 		$plugin            = $plugin . '/' . $plugin_file_names[0];
 	}
 
-	if ( ! current_user_can( 'activate_plugin', $plugin ) ) {
-		wp_die( esc_html__( 'Sorry, you are not allowed to activate this plugin.', 'default' ) );
-	}
-
-	// Activate the plugin in question and return to prior screen.
-	$do_plugin_activation = activate_plugins( $plugin );
-	$referer              = wp_get_referer();
-	if ( ! is_wp_error( $do_plugin_activation ) ) {
-		$referer = add_query_arg(
-			array(
-				'activate' => true,
-			),
-			$referer
-		);
+	$result = activate_plugin( $plugin );
+	if ( is_wp_error( $result ) ) {
+		$status['errorMessage'] = $result->get_error_message();
+		wp_send_json_error( $status );
 	}
 
-	if ( wp_safe_redirect( $referer ) ) {
-		exit;
-	}
+	wp_send_json_success( $status );
 }
-add_action( 'admin_action_perflab_activate_plugin', 'perflab_activate_plugin' );
```

Comment:
Similar to my feedback in #1048, why use AJAX instead of the much simpler admin "action URL" implementation? I'd argue here this is even more relevant, since the previous code was using that approach, and I don't see a benefit for changing this here.

Using AJAX makes the PR unnecessarily complex. I would advise to go back to a regular admin action, and modify the existing action to both install and activate, rather than implementing a whole new JavaScript asset and JSON response behavior. Last but not least, using AJAX also comes with a higher risk that something else in the UI of the page is missed and not updated, so that's another reason to go with the simpler "action URL" approach.

Reply:
Initially, I thought the AJAX approach would work for these changes. However, based on your direction, I updated the approach.